OMCompiler/Compiler/Template/CodegenCpp.tpl

```diff
@@ -13131,51 +13128,37 @@ case _ then
- "Skips generation of defines for jacobian vars."
-::=
- ''
-end variableDefinitionsJacobians_skip;
-
-
 template variableDefinitionsJacobians(list<JacobianMatrix> JacobianMatrixes, SimCode simCode, Text& extraFuncs, Text& extraFuncsDecl, Text extraFuncsNamespace, Text &jacobianVarsInit, Boolean createDebugCode)
  "Generates defines for jacobian vars."
 ::=
-  let analyticVars = (JacobianMatrixes |> JAC_MATRIX(columns=jacColumn, seedVars=vars, matrixName=name, jacobianIndex=jacIndex)  =>
-    let varsDef = variableDefinitionsJacobians2(jacIndex, jacColumn, seedVars, name, simCode, &extraFuncs, &extraFuncsDecl, extraFuncsNamespace, &jacobianVarsInit, createDebugCode)
+if Flags.isSet(NF_SCALARIZE) then
+  ''
+else
+  let analyticVars = (JacobianMatrixes |> JAC_MATRIX(__) =>
+    let seedVarsResult = (seedVars |> var => jacobianVarDefine(var, &jacobianVarsInit, createDebugCode)
+      ;separator="\n";empty)
+    let columnVarsResult = (columns |> JAC_COLUMN(__) =>
+        (columnVars |> var => jacobianVarDefine(var, &jacobianVarsInit, createDebugCode)
+        ;separator="\n";empty)
+      ;separator="\n\n")
     <<
-    <%varsDef%>
+    /*seed vars <%matrixName%>*/
+    <%seedVarsResult%>
+    /*other jac vars <%matrixName%>*/
+    <%columnVarsResult%>
     >>
     ;separator="\n";empty)
-
-    <<
-    /* Jacobian Variables */
-    <%analyticVars%>
-    >>
+  <<
+  /* Jacobian Variables */
+  <%analyticVars%>
+  >>
 end variableDefinitionsJacobians;
 
-template variableDefinitionsJacobians2(Integer indexJacobian, list<JacobianColumn> jacobianColumn, list<SimVar> seedVars, String name, SimCode simCode, Text& extraFuncs, Text& extraFuncsDecl, Text extraFuncsNamespace, Text& jacobianVarsInit, Boolean createDebugCode)
- "Generates Matrixes for Linear Model."
-::=
-  let seedVarsResult = (seedVars |> var hasindex index0 =>
-    jacobianVarDefine(var, indexJacobian, index0, name, simCode, &extraFuncs, &extraFuncsDecl, extraFuncsNamespace, &jacobianVarsInit, createDebugCode)
-    ;separator="\n";empty)
-  let columnVarsResult = (jacobianColumn |> JAC_COLUMN(columnVars=vars) =>
-      (vars |> var hasindex index0 => jacobianVarDefine(var, indexJacobian, index0, name, simCode, &extraFuncs, &extraFuncsDecl, extraFuncsNamespace, &jacobianVarsInit, createDebugCode)
-      ;separator="\n";empty)
-    ;separator="\n\n")
-
-<<
-<%seedVarsResult%>
-<%columnVarsResult%>
->>
-end variableDefinitionsJacobians2;
-
-
-template jacobianVarDefine(SimVar simVar, Integer indexJac, Integer index0, String matrixName, SimCode simCode, Text& extraFuncs, Text& extraFuncsDecl, Text extraFuncsNamespace, Text& jacobianVarsInit, Boolean createDebugCode)
+template jacobianVarDefine(SimVar simVar, Text& jacobianVarsInit, Boolean createDebugCode)
 ""
 ::=
 match simVar
+  /*
```

Comment:
If it's not used any more remove it. If it still has some purpose add some comment on why it is not used now.